src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -159,7 +165,40 @@ impl CatalogState {
                         Datum::from(role.attributes.inherit),
                     ]),
                     diff,
-                })
+                };
+                let mut updates = vec![role_update];
+
+                // HACK/TODO(parkmycar): Creating an empty SessionVars like this is pretty hacky,
+                // we should instead have a static list of all session vars.
+                let session_vars_reference = SessionVars::new_unchecked(
+                    &mz_build_info::DUMMY_BUILD_INFO,
+                    SYSTEM_USER.clone(),
+                );
+
+                for (name, val) in role.vars() {
+                    let result = session_vars_reference
+                        .inspect(name)
+                        .and_then(|var| var.check(val.borrow()));
+                    let formatted_val = match result {
+                        Ok(val) => val,
+                        Err(err) => {
+                            tracing::error!(?name, ?val, ?err, "found invalid role default var");
```

Comment:
Subjective, so feel free to ignore, maybe add a comment indicating that we should have already validated the parameter value by this point.

Reply:
Good call, I added this